build.gradle

```diff
@@ -406,62 +346,91 @@ task packaging(type: Zip) {
 
 defaultTasks 'clean', 'packaging'
 
-/************
- * JavaDoc  *
- ************/
-
 javadoc {
     options.charSet = 'UTF-8'
     options.encoding = 'UTF-8'
 
-    failOnError = true
-
-    source += project(':hivemq-extension-sdk').sourceSets.main.allJava
-
     options.addStringOption('Xdoclint:none', '-quiet')
     options.addStringOption('-html5')
     options.addBooleanOption '-no-module-directories', true
 
     doLast {
         javaexec {
             main = '-jar'
-            args = ["${rootDir}/gradle/tools/javadoc-cleaner-1.0.jar"]
+            args = ["${projectDir}/gradle/tools/javadoc-cleaner-1.0.jar"]
         }
     }
 }
 
-/*******************
- * OWASP CVE CHECK *
- *******************/
+
+/* ******************** checks ******************** */
+
+ext {
+    jacocoVersion = '0.8.5'
+    pmdVersion = '6.22.0'
+    spotbugsVersion = '3.1.12'
+}
+
+compileJava {
+    if (!System.properties['java.version'].startsWith('11')) {
+        throw new RuntimeException('Incompatible JRE version: ' + System.properties['java.version'] + '. Use JRE 11 instead.')
+    }
+}
+
+jacoco {
+    toolVersion = jacocoVersion
+}
+
+pmd {
+    toolVersion = pmdVersion
+    sourceSets = [sourceSets.main]
+    ignoreFailures = true
+    rulePriority = 3
+}
+
+spotbugs {
+    toolVersion = spotbugsVersion
+    sourceSets = [sourceSets.main]
+    ignoreFailures = true
+    reportsDir = file("${buildDir}/reports/findbugs")
+    reportLevel = 'medium'
+}
 
 dependencyCheck {
     analyzers {
         centralEnabled = false
     }
     format = 'ALL'
     scanConfigurations = ['compile']
-    suppressionFile = "${rootDir}/gradle/dependency-check/suppress.xml"
+    suppressionFile = "${projectDir}/gradle/dependency-check/suppress.xml"
     failOnError = false
 }
 check.dependsOn dependencyCheckAnalyze
 
-/*************
- * MiscStuff *
- *************/
+forbiddenApis {
+    bundledSignatures = ['jdk-system-out']
```

Comment:
is sytem err included here?

Reply:
I don't know, this was not changed by me, just moved around the build.gradle file

Reply:
Then its included 👍